nucypher/policy/identity.py

```diff
@@ -0,0 +1,402 @@
+            payload += VariableLengthBytestring(self.__nickname)
+        return payload
+
+    def __hex__(self) -> str:
+        return self.to_hex()
+
+    @property
+    def __payload(self) -> bytes:
+        elements = [
+            self.__character_flag,
+            self.__verifying_key,
+        ]
+        if self.character is Bob:
+            elements.append(self.__encrypting_key)
+        payload = b''.join(bytes(e) for e in elements)
+        return payload
+
+    @classmethod
+    def from_bytes(cls, card_bytes: bytes) -> 'Card':
+        if len(card_bytes) > cls.__MAX_CARD_LENGTH:
+            raise cls.InvalidCard(f'Card exceeds maximum size (max is {cls.__MAX_CARD_LENGTH} bytes card is {len(card_bytes)} bytes). '
+                                  f'Verify the card filepath and contents.')
+        character_flag = card_bytes[:8]
+        if character_flag == bytes(ALICE):
+            specification = cls._alice_specification
+        elif character_flag == bytes(BOB):
+            specification = cls._bob_specification
+        else:
+            raise RuntimeError(f'Unknown character card header ({character_flag}).')
+        return BytestringKwargifier(cls, **specification)(card_bytes)
+
+    @classmethod
+    def from_hex(cls, hexdata: str):
+        return cls.from_bytes(bytes.fromhex(hexdata))
+
+    def to_hex(self) -> str:
+        return bytes(self).hex()
+
+    @classmethod
+    def from_base64(cls, b64data: str):
+        return cls.from_bytes(base64.urlsafe_b64decode(b64data))
+
+    def to_base64(self) -> str:
+        return base64.urlsafe_b64encode(bytes(self)).decode()
+
+    def to_qr_code(self):
+        import qrcode
+        from qrcode.main import QRCode
+        qr = QRCode(
+            version=1,
+            box_size=1,
+            border=4,  # min spec is 4
+            error_correction=qrcode.constants.ERROR_CORRECT_L,
+        )
+        qr.add_data(bytes(self))
+        qr.print_ascii()
+
+    @classmethod
+    def from_dict(cls, card: Dict):
+        instance = cls(nickname=card.get('nickname'),
+                       verifying_key=card['verifying_key'],
+                       encrypting_key=card['encrypting_key'],
+                       character_flag=card['character'])
+        return instance
+
+    def to_dict(self) -> Dict:
+        payload = dict(
+            nickname=self.__nickname,
+            verifying_key=self.verifying_key,
+            encrypting_key=self.encrypting_key,
+            character=self.__character_flag
+        )
+        return payload
+
+    def describe(self, truncate: int = TRUNCATE) -> Dict:
+        description = dict(
+            nickname=self.__nickname,
+            id=self.id.hex(),
+            verifying_key=bytes(self.verifying_key).hex()[:truncate],
+            character=self.character.__name__
+        )
+        if self.character is Bob:
+            description['encrypting_key'] = bytes(self.encrypting_key).hex()[:truncate],
```

Comment:
I'm assuming not meant to be a tuple:
```bash
 [47] → nucypher contacts list
    | nickname   | id           | verifying_key    | character   | encrypting_key
----+------------+--------------+------------------+-------------+-----------------------
  0 | Rando      | 0x0b715812e1 | 027b8c9cf91929e0 | Alice       |
  1 | BobRando   | 0x5aeab852bb | 027b8c9cf91929e0 | Bob         | ('0283c908f7ad4d61',)
```


```suggestion
            description['encrypting_key'] = bytes(self.encrypting_key).hex()[:truncate]
```

Reply:
Heh - extra comma in the `describe` method. :heavy_check_mark: